test/end-to-end/offer.js

```diff
@@ -603,4 +604,71 @@ describe("Offer endpoint tests", () => {
             });
         });
     });
+
+    describe("GET /offers/offerId", () => {
+
+        beforeAll(async () => {
+            await Offer.deleteMany({});
+        });
+
+        describe("Id Validation", () => {
+            test("Invalid ID", async () => {
+                const res = await request()
+                    .get("/offers/new/123");
+
+                expect(res.status).toBe(HTTPStatus.NOT_FOUND);
+                expect(res.body).toHaveProperty("reason", ValidationReasons.OBJECT_ID);
+            });
+            test("Offer not found", async () => {
+                const id = "5facf0cdb8bc30016ee58952";
+                const res = await request()
+                    .get(`/offers/new/${id}`);
+
+                expect(res.status).toBe(HTTPStatus.NOT_FOUND);
+                expect(res.body).toHaveProperty("reason", APIErrorTypes.OFFER_NOT_FOUND(id));
+            });
+        });
+
+        describe("Get offer by Id", () => {
+            const test_offer_1 = {
+                title: "Test Offer 1",
+                publishDate: "2021-01-14T00:00:00.000Z",
+                publishEndDate: "20-02-28T00:00:00.000Z",
+                description: "For Testing Purposes",
+                contacts: ["geral@niaefeup.pt", "229417766"],
+                jobType: "SUMMER INTERNSHIP",
+                fields: ["DEVOPS", "MACHINE LEARNING", "OTHER"],
+                technologies: ["React", "CSS"],
+                // owner: Will be set in beforeAll,
+                location: "Testing Street, Test City, 123",
+            };
+            const test_offer_2 = {
+                title: "Test Offer",
+                publishDate: "2019-11-22T00:00:00.000Z",
+                publishEndDate: "2019-11-28T00:00:00.000Z",
+                description: "For Testing Purposes",
+                contacts: ["geral@niaefeup.pt", "229417766"],
+                jobType: "SUMMER INTERNSHIP",
+                fields: ["FRONT_END", "BACK_END"],
+                technologies: ["Flutter", "Node.js"],
+                // owner: Will be set in beforeAll,
+                location: "Testing Avenue, Test State, 321",
+            };
+            beforeAll(async () => {
+                await Offer.create([test_offer_1, test_offer_2]);
+            });
+
+            test("Offer 1", async () => {
+                const res = await request().get(`/offers/${test_offer_1.id}`);
+                expect(res.status).toBe(HTTPStatus.OK);
+                expect(res.body).toBe(test_offer_1);
+            });
+
+            test("Offer 1", async () => {
+                const res = await request().get(`/offers/${test_offer_1.id}`);
+                expect(res.status).toBe(HTTPStatus.OK);
+                expect(res.body).toBe(test_offer_1);
+            });
```

Comment:
This should only be 1 test, since it is the same test. Just update the title and merge the two tests (basically making the two requests and the `expects` of the first between them